### DISCUSSION TITLE
What is the best approach for multiple API endpoints?

### DISCUSSION THREAD

**Discussion by onurkose** [Akryum/vue-apollo#284] [Q&A] [answered]
Hello,
I already know Apollo has Multiple Clients functionality. But my question is how to handle when a single API has multiple endpoints;

This is the default endpoint:
http://127.0.0.1:8000/graphql

This is secret endpoint;
http://127.0.0.1:8000/graphql/restricted

Maybe some other endpoint whether or not secret;
http://127.0.0.1:8000/graphql/countries

Apollo Http Link has [context](https://github.com/apollographql/apollo-link/tree/master/packages/apollo-link-http) feature so I can also change the endpoint on the go;

```javascript
new Vue({
  apollo: {
    serverData: {
      query: gql`query getCountries($status:String!) {
        countries (status: $status) {
          id,
          name
        }
      }`,
      variables: {
        status: 'active'
      },
      context: {
        uri: 'http://127.0.0.1:8000/graphql/countries'
      }
    }
  },
})
```

What I understand from the Multiple Clients, I can also use this setup;

```javascript
const apolloProvider = new VueApollo({
  clients: {
    default: apolloClient,
    secret: secretApolloClient,
    countries: countriesApolloClient
  },
  defaultClient: apolloClient,
})

new Vue({
  apollo: {
    serverData: {
      query: gql`query getCountries($status:String!) {
        countries (status: $status) {
          id,
          name
        }
      }`,
      variables: {
        status: 'active'
      },
      $client: 'countries'
    }
  },
})
```

So what's the true approach here? Am I mixing apples and pears?

**Comment by Akryum** [comment, marked as answer]
I would say both are valid, so use what suits you best.

**Comment by onurkose** [comment]
@Akryum Thank you for prompt enlightenment!
